app/src/main/java/com/google/android/apps/youtube/app/settings/videoquality/VideoQualitySettingsActivity.java

```diff
@@ -75,6 +77,56 @@ protected void onCreate(Bundle bundle) {
 
             // Set search view
             SearchView searchView = findViewById(ResourceUtils.getIdIdentifier("search_view"));
+
+            // region SearchView dimensions
+            // Get the current layout parameters of the SearchView
+            ViewGroup.MarginLayoutParams layoutParams = (ViewGroup.MarginLayoutParams) searchView.getLayoutParams();
+
+            // Set the margins (in pixels)
+            int margin = (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, 20, getResources().getDisplayMetrics()); // for example, 10dp
+            layoutParams.setMargins(margin, layoutParams.topMargin, margin, layoutParams.bottomMargin);
+
+            // Apply the layout parameters to the SearchView
+            searchView.setLayoutParams(layoutParams);
+
+            // endregion
+
+            // region SearchView color
+
+            GradientDrawable shape = new GradientDrawable();
+
+            String currentDarkHex = DrawableColorPatch.getDarkBackgroundHexValue();
+            String currentLightHex = DrawableColorPatch.getLightBackgroundHexValue();
+
+            String currentHex = ThemeUtils.isDarkTheme() ? currentDarkHex : currentLightHex;
+            String defaultHex = ThemeUtils.isDarkTheme() ? "#1A1A1A" : "#E5E5E5";
+
+            String finalHex;
+            if (currentHex.equals(ThemeUtils.isDarkTheme() ? "#000000" : "#FFFFFF")) {
+                shape.setColor(Color.parseColor(defaultHex)); // stock black/white color
+                finalHex = defaultHex;
+            } else {
+                // custom color theme
+                String adjustedColor = ThemeUtils.isDarkTheme()
+                        ? ThemeUtils.lightenColor(currentHex, 15)
+                        : ThemeUtils.darkenColor(currentHex, 15);
+                shape.setColor(Color.parseColor(adjustedColor));
+                finalHex = adjustedColor;
+            }
+            Logger.printInfo(() -> "searchbar color: " + finalHex);
+
+            shape.setCornerRadius(30 * getResources().getDisplayMetrics().density);
+            searchView.setBackground(shape);
+
+            /*
+             * in order to match the original app's search bar, we'd need to change the searchbar cursor color
+             * to white (#FFFFFF) if ThemeUtils.isDarkTheme(), and black (#000000) if not.
+             */
+
+            // endregion
+
+            searchView.setPadding(20, 15, 20, 15);
```

Comment:
dp value is not used for padding
Is this intentional?

Reply:
I removed 
```xml
android:padding="5dp" />
```
from the xml file, and did all the padding at runtime. It seemed like the easiest solution to me

Reply:
If raw value is used instead of dp, different values ​​may be applied depending on the resolution of the device

Since dp was used as a padding value in the original xml (`android:padding="5dp" />`), it is better to use dp when setting padding programmatically

To convert raw value ​​to dp value ​​you can use this syntax:
https://github.com/inotia00/revanced-integrations/blob/dc0a5e1de828260c1fc6dbfc494abc004a7a7987/app/src/main/java/com/google/android/apps/youtube/app/settings/videoquality/VideoQualitySettingsActivity.java#L86

Reply:
ah, I thought they were converted to DP at the end. my bad, I'll convert them to DP and test the final result

Reply:
resolved in 2f702ac